plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceRegistry.java

```diff
@@ -170,7 +173,16 @@ DBPDataSourceConfigurationStorage getDefaultStorage() {
                 }
             }
             // No default storage. Seems to be an internal error
-            throw new IllegalStateException("no default storage in registry " + this);
+            log.warn("no default storage in registry " + this);
+            try {
+                java.nio.file.Path configPath = this.getProject().getMetadataFolder(false).resolve(MODERN_CONFIG_FILE_NAME);
+                Files.createFile(configPath);
+                DBPDataSourceConfigurationStorage defaultStorage = new DataSourceFileStorage(configPath, false, true);
+                this.storages.add(defaultStorage);
```

Comment:
Looks fragile and hack-ish. I'm not sure 😐

Reply:
We already add default storage in DataSourceConfigurationManagerNIO::getConfigurationStorages() if there is no one. I guess someone just forgot to create a new file for this case.
```
if (storages.isEmpty()) {
    storages.add(
        new DataSourceFileStorage(
            metadataFolder.resolve(DataSourceRegistry.MODERN_CONFIG_FILE_NAME), false, true));
}
```
Anyway I see default storage everywhere. If I correctly understand the code, it wasn't assumed that we could not have default storage or this case wasn't correctly handled.

Reply:
Maybe I could handle it there.